File: python/bob/bob.py
import re


def hey(phrase):
    ''' test phrase to determine output  '''
    only_whitespace = re.search(r'^\s+$', phrase)

    question = re.search(r'.+\?\s*$', phrase)
    q_all_caps_with_whitespace = re.search(r'^[A-Z\s]+\?\s*$', phrase)
    numbers_comma_whitespace = re.search(r'^[0-9,\s]+$', phrase)
    everything_not_lower_case = re.search(r'^[^a-z]+[\!\s]*$', phrase)

    if only_whitespace or len(phrase) == 0:
        return 'Fine. Be that way!'
    elif q_all_caps_with_whitespace:
        return "Calm down, I know what I'm doing!"
    elif question:
        return 'Sure.'
    elif everything_not_lower_case and not numbers_comma_whitespace:
        return 'Whoa, chill out!'
    else:
        return 'Whatever.'

File: go/bob/bob.go
// bob is a chat bot
// use this instead: https://golang.org/pkg/strings/
package bob

import "regexp"

// Hey takes a comment and returns a response
func Hey(remark string) string {

	switch {

	case regexp.MustCompile(`^\s+$`).MatchString(remark) || len(remark) == 0:
		return "Fine. Be that way!"

	case regexp.MustCompile(`^[A-Z\s]+\?\s*$`).MatchString(remark):
		return "Calm down, I know what I'm doing!"

	case regexp.MustCompile(`.+\?\s*$`).MatchString(remark):
		return "Sure."

	case regexp.MustCompile(`^[0-9,\s]+$`).MatchString(remark):
		return "Whatever."

	case regexp.MustCompile(`^[^a-z]+[\!\s]*$`).MatchString(remark):
		return "Whoa, chill out!"

	default:
		return "Whatever."

	}

}

// https://golang.org/doc/effective_go.html#commentary
